src/find/matchers/samefile.rs

```diff
@@ -0,0 +1,60 @@
+// This file is part of the uutils findutils package.
+//
+// For the full copyright and license information, please view the LICENSE
+// file that was distributed with this source code.
+
+use std::path::PathBuf;
+
+use super::Matcher;
+
+pub struct SameFileMatcher {
+    path: PathBuf,
+}
+
+impl SameFileMatcher {
+    pub fn new(path: PathBuf) -> SameFileMatcher {
+        SameFileMatcher { path }
+    }
+}
+
+impl Matcher for SameFileMatcher {
+    #[cfg(unix)]
+    fn matches(&self, file_info: &walkdir::DirEntry, _matcher_io: &mut super::MatcherIO) -> bool {
+        uucore::fs::paths_refer_to_same_file(file_info.path(), &self.path, true)
+    }
+
+    #[cfg(not(unix))]
+    fn matches(&self, _file_info: &walkdir::DirEntry, _matcher_io: &mut super::MatcherIO) -> bool {
+        // FIXME
```

Comment:
please create an issue for this once it landed

Reply:
also, it supports windows?
https://github.com/uutils/coreutils/blob/92665144c9ddf100d5044dc0c52af122a94587d0/src/uu/split/src/platform/windows.rs#L47

Reply:
> also, it supports windows? https://github.com/uutils/coreutils/blob/92665144c9ddf100d5044dc0c52af122a94587d0/src/uu/split/src/platform/windows.rs#L47

Oh! That's right! Changed to support multiple platforms. Thanks!